src/components/modals/ModalWithActionButtons.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { Button, ButtonProps } from '../buttons/Button';
+
+import { BaseModal } from './BaseModal';
+
+export interface ActionButton extends ButtonProps {
+  actionCallback?: (...params: any[]) => any;
+  closeModalOnClick?: boolean;
+}
+
+export interface ModalProps {
+  title: string;
+  contentText: string;
+  open: boolean;
+  handleClose: () => void;
+}
+
+interface ModalActionButtonProps {
```

Comment:
im confused, this is props for the modal with action buttons or the action buttons within the modal itself